cpp/src/Slice/Parser.cpp

```diff
@@ -5784,12 +5734,15 @@ Slice::Operation::uses(const ContainedPtr& contained) const
         }
     }
 
-    for (const auto& exception : _throws)
+    if (ExceptionPtr::dynamicCast(contained))
     {
-        ContainedPtr contained2 = ContainedPtr::dynamicCast(exception);
-        if (contained2 && contained2 == contained)
+        for (const auto& exception : _throws)
         {
-            return true;
+            ContainedPtr contained2 = ContainedPtr::dynamicCast(exception);
```

Comment:
Why don't keep the result of the dynamicCast above and avoid this dynamicCast, just compare the exceptions instead of the contained